tests/test_views.py

```diff
@@ -86,3 +96,103 @@ def test_post_detail_failed(self):
             reverse('post_detail', kwargs={'pk': 31}),
         )
         self.assertEqual(404, http_response.status_code)
+
+    def test_post_new_get_authorized(self):
+        """Ensure authorized users can access the new post page."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        http_response = self.client.get(url)
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_edit.html')
+
+    def test_post_new_get_unauthorized(self):
+        """Ensure unauthorized users are redirected when add the new post."""
+        self.client.logout()
+        url = reverse('post_new')
+        http_response = self.client.get(url)
+        self.assertRedirects(http_response, f'/accounts/login/?next={url}')
+
+    def test_post_new_valid_form_authorized(self):
+        """Ensure authorized users can correctly fill out the form for post."""
+        self.client.force_login(self.user)
+        url = reverse('post_new')
+        initial_post_count = Post.objects.count()
+        http_response = self.client.post(
+            url, self.valid_form_post, follow=True,
+        )
+        self.assertEqual(http_response.status_code, 200)
+        self.assertTemplateUsed(http_response, 'blog/post_detail.html')
+        new_post = Post.objects.last()
+        self.assertRedirects(
+            http_response,
+            reverse(
+                'post_detail',
+                kwargs={'pk': new_post.pk},
+            ),
+        )
+        self.assertEqual(new_post.title, 'Test Title')
```

Comment:
It's a bit difficult to connect the dots when this constant string is defined here, but the data it compares to is defined elsewhere. Why not reuse the same value saved into a var?